src/librustc_resolve/error_codes.rs

```diff
@@ -1660,6 +1660,20 @@ fn const_id<T, const N: T>() -> T { // error: const parameter
 }
 ```
 "##,
+
+E0734: r##"
+Type parameter defaults cannot use `Self` on structs, enums, or unions.
+Erroneous code example:
+
+```compile_fail,E0734
+struct Foo<X=Box<Self>> {
```

Comment:
```suggestion
struct Foo<X = Box<Self>> {
```

Reply:
Okay, I'll change the other diagnostic in the file that uses type parameter defaults to use this formatting too, for consistency.

Reply:
Excellent, thanks. =)